### PR TITLE
Add Direct Line App Service Extension protocol

### LINKED ISSUE #3205: Support Direct Line App Service Extension protocol
## Feature Request

Add support of Direct Line App Service Extension protocol, according to https://docs.microsoft.com/en-us/azure/bot-service/bot-service-channel-directline-extension?view=azure-bot-service-4.0.

[Enhancement]

### PR DESCRIPTION
**Pull request by compulim**
> Fixes #3205.

### Added

-  Resolves [#3205](https://github.com/microsoft/BotFramework-WebChat/issues/3205). Added Direct Line App Service Extension protocol, by [@compulim](https://github.com/compulim) in PR [#3206](https://github.com/microsoft/BotFramework-WebChat/pull/3206)

### Changed

-  Bumped [`botframework-directlinejs@0.12.0`](https://npmjs.com/package/botframework-directlinejs), by [@compulim](https://github.com/compulim) in PR [#3206](https://github.com/microsoft/BotFramework-WebChat/pull/3206)

### Samples

-  Resolves [#3205](https://github.com/microsoft/BotFramework-WebChat/issues/3205). Added [Direct Line App Service Extension chat adapter](https://microsoft.github.io/BotFramework-WebChat/01.getting-started/i.protocol-direct-line-app-service-extension) sample, by [@compulim](https://github.com/compulim) in PR [#3206](https://github.com/microsoft/BotFramework-WebChat/pull/3206)

## Description

Add Direct Line App Service Extension chat adapter.

## Specific Changes

- Bundled the adapter through `createDirectLineAppServiceExtension`
   - Exposed in both bundle and NPM
   - As this is built into DirectLineJS, it is supported in minimal bundle
- Added sample for Direct Line App Service Extension chat adapter
- Bumped to `botframework-directlinejs@0.12.0`
- Removing some tests as they are already migrated to the new HTML-based tests
- Migrated some tests to HTML-based instead of fixing them
   - New `pageObjects.waitWithTick()` to run `lolex.tick(1)` while waiting

---

-  [x] Testing Added

### CODE REVIEW

**Comment by compulim** [comment]
After this is merged, please update https://docs.microsoft.com/en-us/azure/bot-service/bot-service-channel-directline-extension-webchat-client?view=azure-bot-service-4.0.

File: samples/01.getting-started/i.protocol-direct-line-app-service-extension/README.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
+# Sample - Getting Started with Direct Line App Service Extension protocol
```
Comment (corinagum) [inline samples/01.getting-started/i.protocol-direct-line-app-service-extension/README.md:1]:
```suggestion
# Sample - Direct Line App Service Extension protocol
```

File: samples/01.getting-started/i.protocol-direct-line-app-service-extension/README.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+# Code
+
+> Jump to [completed code](#completed-code) to see the end-result `index.html`.
+
+## Getting started
+
+### Goals of this bot
+
+This code features the Direct Line App Service Extension protocol (Direct Line ASE). Additional steps required to set up a bot to use Direct Line App Service Extension can be found [in this documentation](https://docs.microsoft.com/en-us/azure/bot-service/bot-service-channel-directline-extension?view=azure-bot-service-4.0).
```
Comment (corinagum) [inline samples/01.getting-started/i.protocol-direct-line-app-service-extension/README.md:30]:
```suggestion
This code features the Direct Line App Service Extension protocol (Direct Line ASE). Additional steps required to set up a bot to use Direct Line App Service Extension can be found in the [DL ASE documentation](https://docs.microsoft.com/en-us/azure/bot-service/bot-service-channel-directline-extension?view=azure-bot-service-4.0).
```
Links must have accessible text

File: samples/01.getting-started/i.protocol-direct-line-app-service-extension/README.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+The `index.html` page has few main goals.
```
Comment (corinagum) [inline samples/01.getting-started/i.protocol-direct-line-app-service-extension/README.md:32]:
```suggestion
The `index.html` page has the following main goals:
```

File: __tests__/html/offlineUI.sendFailed.noAck.html [hunk 4]
```diff
@@ -9,53 +9,57 @@
...
-
-              type === 'message' && bannedClientActivityIDs.push(clientActivityID);
-
-              return workingDirectLine.postActivity(activity);
-            }
-          };
-        }
+      function allOutgoingMessagesFailed() {
+        return {
+          message: 'All outgoing messages to fail sending',
```
Comment (corinagum) [inline __tests__/html/offlineUI.sendFailed.noAck.html:14]:
```suggestion
          message: 'All outgoing messages failed to send',
```

**Pull request merged**